### PR TITLE
RRD: use Lighthouse query methods directly

### PR DESCRIPTION
**Pull request by yoomlam**
## Description of change

Have callers call the `list_observations` and now `list_medication_requests` directly, instead of `list_resource(resource)`.

## Original issue(s)
There's little benefit to indirectly calling `list_resource(resource)`. Removing it will clean up and reduce code.
[Slack thread](https://dsva.slack.com/archives/C031RRFCHV1/p1647020879483619)

## Things to know about this PR

No change to code behavior.

### CODE REVIEW

File: spec/lib/lighthouse/veterans_health/integration/client_spec.rb [hunk 1]
```diff
@@ -5,13 +5,29 @@
...
-  describe '#list_resource' do
+  describe '#list_medication_requests' do
     context 'with a multi-page response' do
-      subject(:client) { described_class.new(32_000_225) }
+      subject(:client) { described_class.new(32_000_225).list_medication_requests }
 
       it 'returns all entries in the response' do
-        response = client.list_resource('medication_requests')
-        expect(response.body['entry'].count).to match response.body['total']
+        VCR.use_cassette('lighthouse/veterans_health/medication_requests') do
```
Comment (yoomlam) [inline spec/lib/lighthouse/veterans_health/integration/client_spec.rb:13]: Moved cassette to new (more consistent) location.

File: spec/lib/lighthouse/veterans_health/integration/client_spec.rb [hunk 2]
```diff
@@ -5,13 +5,29 @@
...
+        end
+      end
+    end
+  end
+
+  describe '#list_observations' do
+    subject(:client) { described_class.new(2_000_163).list_observations }
+
+    it 'returns all sorts of observations' do
+      VCR.use_cassette('lighthouse/veterans_health/observations') do
```
Comment (yoomlam) [inline spec/lib/lighthouse/veterans_health/integration/client_spec.rb:24]: Added new cassette for new test.

**Review by leikkisa — APPROVED** [review]
This looks great 👍

**Review by ericboehs — APPROVED** [review]

**Review by allthesignals — APPROVED** [review]
Thanks. Looks good. Seems like it was meant to become an ORM wrapper for Lighthouse. This is more appropriate.

**Pull request merged**